src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -127,14 +127,19 @@ def validator(ns):
 
 def validate_key_import_source(ns):
     byok_file = ns.byok_file
+    byok_string = ns.byok_string
     pem_file = ns.pem_file
+    pem_string = ns.pem_string
     pem_password = ns.pem_password
-    if (not byok_file and not pem_file) or (byok_file and pem_file):
-        raise ValueError('supply exactly one: --byok-file, --pem-file')
-    if byok_file and pem_password:
-        raise ValueError('--byok-file cannot be used with --pem-password')
-    if pem_password and not pem_file:
-        raise ValueError('--pem-password must be used with --pem-file')
+    if not ((byok_file and not byok_string and not pem_file and not pem_string) or
+            (not byok_file and byok_string and not pem_file and not pem_string) or
+            (not byok_file and not byok_string and pem_file and not pem_string) or
+            (not byok_file and not byok_string and not pem_file and pem_string)):
```

Comment:
As we have 4 args now, it's not a good way to list all the conditions. This is the recommended implementation:
```python
if len([arg for arg in [byok_file, byok_string, pem_file, pem_string] if arg]) != 1:
```